src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -2524,18 +2564,86 @@ bool StorageReplicatedMergeTree::executeReplaceRange(const LogEntry & entry)
         }
     }
 
-    static const String TMP_PREFIX = "tmp_replace_from_";
-
     auto obtain_part = [&] (PartDescriptionPtr & part_desc)
     {
         if (part_desc->src_table_part)
         {
-
             if (part_desc->checksum_hex != part_desc->src_table_part->checksums.getTotalChecksumHex())
                 throw Exception("Checksums of " + part_desc->src_table_part->name + " is suddenly changed", ErrorCodes::UNFINISHED);
 
-            part_desc->res_part = cloneAndLoadDataPartOnSameDisk(
-                part_desc->src_table_part, TMP_PREFIX + "clone_", part_desc->new_part_info, metadata_snapshot);
+            if (entry_replace.from_table.empty())
+            {
+                String partition_id = part_desc->new_part_info.partition_id;
+                FutureMergedMutatedPart future_part;
+                future_part.parts.push_back(part_desc->src_table_part);
+                future_part.part_info = part_desc->new_part_info;
+                future_part.name = part_desc->new_part_name;
+                future_part.type = part_desc->src_table_part->getType();
+                auto reservation = tryReserveSpace(
+                    MergeTreeDataMergerMutator::estimateNeededDiskSpace({part_desc->src_table_part}), part_desc->src_table_part->volume);
+                MergeList::EntryPtr merge_entry = getContext()->getMergeList().insert(getStorageID(), future_part);
+                Stopwatch stopwatch;
+                MutableDataPartPtr new_part;
+
+                auto write_part_log = [&] (const ExecutionStatus & execution_status)
+                {
+                    writePartLog(
+                        PartLogElement::MUTATE_PART,
+                        execution_status,
+                        stopwatch.elapsed(),
+                        future_part.name,
+                        new_part,
+                        future_part.parts,
+                        merge_entry.get());
+                };
+
+                try
+                {
+                    auto replace_commands = entry_replace.commands;
+                    if (replace_commands.empty())
+                        throw Exception("There is no command to update. This is a bug", ErrorCodes::LOGICAL_ERROR);
+
+                    auto & update = replace_commands[0];
+                    Names minmax_idx_columns = getMinMaxColumnsNames(metadata_snapshot->getPartitionKey());
+                    std::unordered_set<String> columns(minmax_idx_columns.begin(), minmax_idx_columns.end());
+                    for (const auto & [name, _] : update.column_to_update_expression)
+                        columns.erase(name);
+
+                    // Fill in missing minmax columns when mutating partition key
+                    if (!columns.empty())
+                    {
+                        for (const auto & name : columns)
+                            update.column_to_update_expression.emplace(name, std::make_shared<ASTIdentifier>(name));
```

Comment:
Also it looks like a hack. Is it possible to understand that we need to update these columns without adding redundant expressions into mutation commands?